src/Documentation/Getting-Started-With-Orleans/Grain-LifeCycle.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: page
+title: Grain LifeCycle
+---
+
+
+## Grain LifeCycle
+
+We say grains virtually always exist but here we want to describe different stages of a grain's life.
+A grain is like a class definition, it is nothing unless it is on the memory and can execute code.
+An instance of a class is instantiated with the `new` keyword in C#.
+an instance of a grain is called an activation.
+The grain's instances are differentiated by their key and a grain with a specific key is either activated or not.
+You don't control this by using construction/destruction keywords and instead Orleans runtime manages this for you.
+
+All possible grains in the whole key space virtually always exist
+and it means whenever you call a method of a grain, if it is not active, it will get activated by the runtime.
+Also after the grain passes a certain amount of time in idle state, the runtime will deactivate the grain (i.e. destroys it).
+A grain is in idle state if it is not currently processing a message and its queue of messages is empty.
+
+When you get a grain reference in a client or in another grain by using `GetGrain`,
+you only get a reference to a grain and you don't get the activation.
+When you call a method using the reference,then the grain will be activated in a silo (if it is not already active)
+and your reference sends messages to the activation by calling methods and points to the activation logically.
+
+`GetGrain` itself is not asynchronous, can not fail and only returns the logical reference which points to the activation.
+Resolving the reference->activation map is one of the many things that runtime does for you 
+
+## callbacks 
+
+Just like the class which can have constructors and (destructors in some languages), grains have
+`OnActivateAsync` and `OnDeactivateAsync` virtual methods which you can override.
+
+## Different grain types and their differences regarding life cycle
+
+Your grains can be persistent and store their state in storage
+or have no state which is stored in storage. 
+In case of the later, all activations are created equal and there is no difference between them but in case of the former
+each activation reads the state from storage so the `State` contains latest stored state.
+For grains which store their state and derive from `Grain<T>` there is a point which you should consider.
+
+> `ReadStateAsync` is called before calling `OnActivateAsync` and if it fails then the grain will not be activated.
+
+
+## Sequence of events in a grain's life cycle
+
+So the life cycle of a grain is like this
+
+- another grain or a client calls a method of a grain
+- the grain gets activated and brought to memory
```

Comment:
I would clarify explicitly "if the grain is not activated anywhere in the cluster."
Then I would say that "an instance of the grain class gets created" instead of "grain gets activated and brought to memory" to be more precise.